File: 프로그래머스/Lv.1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    /*
    completion.forEach((value, index) => {
        let key = participant.findIndex(element => element === value)
        participant.splice(key, 1)
    })
    return participant[0]
    */
    let answer = ""
    participant.sort()
    completion.sort()
    for(let i=0; i<completion.length; i++) {
        if(participant[i] !== completion[i]) {
            answer = participant[i]
            break;
        }
    }
    console.log(answer === ""? participant[participant.length] : answer )
    return answer === ""? participant[participant.length-1] : answer 
    
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    map<string, int> racePlayer;
    for(string s: participant) {
        racePlayer[s] += 1;
    }
    
    for(string s: completion) {
        racePlayer[s] -= 1;
    }
    
    for(auto player: racePlayer) {
        if(player.second > 0) {
            return player.first;
        }
    }
    
    return answer;
}